user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md

```diff
@@ -0,0 +1,199 @@
+
+> [!NOTE]
+> The changes mentioned on this page require **DataMiner 10.5.1/10.5.0 or higher**. Note that while legacy alarm references may still be used in DataMiner 10.5.1 or higher, they must not be used on systems where the Swarming feature is enabled.
+
+## About the Alarm ID changes
+
+In order to implement the necessary changes to prepare for Swarming, you first need to understand what the alarm ID changes are that get introduced with Swarming and what their consequences are.
+
+For this purpose, you need to be familiar with the following terminology:
+
+- **Alarm event**: An object indicating that either a new alarm came up, an existing event was updated (severity changed, properties updated, acknowledged, etc.), or an existing event was cleared.
+- **Root event**: Initial alarm event, when an alarm is created where previously there was no alarm.
+- **Alarm tree**: A logical alarm, combining a history of alarm events for the same alarm. A tree will always have a root alarm event. Any alarm event updates that have happened over time will be added to this. Each update links back to the original root alarm event as well as to the previous alarm event and has a unique alarm ID number (e.g. `444`) within that tree.
+- **Alarm tree ID**: Reference that can be used to uniquely refer to an alarm tree without having to refer to individual alarm events. For example, a comment is added to an alarm tree, the alarm tree ID can be sent to the server to apply the update to the most recent state of that alarm tree. An ID has the format `111/222/333`, where `111/222` is a reference to the element involved. The Alarm tree ID is the preferred way to reference alarm trees.
+- **Root alarm ID**: Legacy number (e.g. `333`) indicating a specific alarm tree. This is not guaranteed to be unique across elements or DMAs.
+- **Alarm ID**: Number referencing an alarm event within an alarm tree (e.g. `444`). This should not be used on its own, but it can be used in combination with an alarm tree ID in cases where an individual alarm event needs to be referenced. Try to avoid such alarm ID references ([more info](#alarmid-references)). For root alarm events, the alarm ID is identical to the root alarm ID.
+- **Hosting Agent ID**: The DataMiner ID of the DMA hosting the element that the alarm event/tree is linked to.
+
+In DataMiner versions prior to DataMiner 10.4, the combination `HostingAgentID/RootAlarmID` is typically used to refer to an alarm tree. When elements are [migrated between DMAs](xref:Migrating_elements_in_a_DataMiner_System) or swarmed from one DMA to another, this combination of fields is inadequate as it is not unique or fixed.
```

Comment:
Which DataMiner versions introduce this change exactly? Is there an RN reference we can add here?